scipy/spatial/_qhull.pyx

```diff
@@ -2870,6 +2870,15 @@ class HalfspaceIntersection(_QhullUser):
         of halfspaces is also not possible after `close` has been called.
 
         """
+        # We check for non-feasibility of incremental additions
+        # in a manner similar to `qh_sethalfspace`
+        dims = halfspaces.shape[1] - 1
+        for halfspace in halfspaces:
```

Comment:
Our testsuite may not be sensitive to it, but now that I look at this, I suspect I/we should check for the case where a single violating halfspace is added incrementally rather than a bunch of them. My intuition is we may get the wrong behavior in that case.

Reply:
I just mean that if you feed in a 1d array like `[0.4, 0.4, 0.3]` for the new halfspace that line might iterate over the elements instead of the half spaces in a 2d array as currently tested.